src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -209,6 +218,17 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                     isSolutionAvailable,
                     deferredProjectsData.PackageSpecs,
                     token);
+
+                // To limit risk, we only publish the event when there is a cross-platform PackageReference project
+                // in the solution. Extending this behavior to all solutions is tracked here:
+                // https://github.com/NuGet/Home/issues/4478
+#if VS15
+                if (projects.OfType<CpsPackageReferenceProject>().Any())
+                {
+                    var args = new SolutionRestoredEventArgs(_dependencyGraphProjectCacheHash);
+                    _restoreEventsPublisher.OnSolutionRestoreCompleted(args);
+                }
```

Comment:
I opted for sending an `isSuccess` bool so that PowerShell does not have to think about all the different state transitions. We can revisit this decision in the future when other subscribers listen to this event.